packages/workspace/src/schematics/remove/remove.ts

```diff
@@ -0,0 +1,17 @@
+import { chain, Rule } from '@angular-devkit/schematics';
+import { checkImports } from './lib/check-imports';
+import { checkTargets } from './lib/check-targets';
+import { removeProject } from './lib/remove-project';
+import { updateNxJson } from './lib/update-nx-json';
+import { updateWorkspace } from './lib/update-workspace';
+import { Schema } from './schema';
+
+export default function(schema: Schema): Rule {
+  return chain([
+    checkImports(schema),
```

Comment:
What if instead of reinventing the dependency logic, we use the dep-graph in here and make sure that the project is a stray node.

Reply:
I'll take a look at it - I wasn't aware it was usable for this kind of case.

Reply:
I'm going to add this comment here just as a record - 

As per Slack conversation, I'm going to factor out the file reading code (which may not be accurate enough) and use `createProjectGraph` to determine the actual dependencies so that we're definitely consistent. We'll live with the possible innaccuracy of the file output for now.